examples/camera-app/camera-common/BUILD.gn

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2025 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+import("${chip_root}/src/app/chip_data_model.gni")
+
+config("config") {
+  include_dirs = [ "include" ]
+}
+
+chip_data_model("camera-common") {
+  zap_file = "camera-app.zap"
+  is_server = true
+}
```

Comment:
This folder contains the platform independent portion of the app, we need to define a source_set "camera-lib" to build camera-app.h/cpp here instead of each platform

Reply:
Created a source-set for the camera-app files and headers.